dolphinscheduler-storage-plugin/dolphinscheduler-storage-cos/src/main/java/org/apache/dolphinscheduler/plugin/storage/cos/CosStorageOperator.java

```diff
@@ -0,0 +1,335 @@
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import com.qcloud.cos.COSClient;
+import com.qcloud.cos.ClientConfig;
+import com.qcloud.cos.auth.BasicCOSCredentials;
+import com.qcloud.cos.auth.COSCredentials;
+import com.qcloud.cos.exception.CosServiceException;
+import com.qcloud.cos.http.HttpProtocol;
+import com.qcloud.cos.model.COSObject;
+import com.qcloud.cos.model.COSObjectSummary;
+import com.qcloud.cos.model.CopyObjectRequest;
+import com.qcloud.cos.model.CopyResult;
+import com.qcloud.cos.model.GetObjectRequest;
+import com.qcloud.cos.model.ListObjectsRequest;
+import com.qcloud.cos.model.ObjectListing;
+import com.qcloud.cos.model.ObjectMetadata;
+import com.qcloud.cos.model.PutObjectRequest;
+import com.qcloud.cos.model.UploadResult;
+import com.qcloud.cos.region.Region;
+import com.qcloud.cos.transfer.Copy;
+import com.qcloud.cos.transfer.Download;
+import com.qcloud.cos.transfer.TransferManager;
+import com.qcloud.cos.transfer.TransferManagerConfiguration;
+import com.qcloud.cos.transfer.Upload;
+
+@Slf4j
+public class CosStorageOperator extends AbstractStorageOperator implements Closeable, StorageOperator {
+
+    private static final int TRANSFER_THREAD_POOL_SIZE = 16;
+
+    private static final long MULTIPART_UPLOAD_BYTES_THRESHOLD = 5 * 1024 * 1024L;
+    private static final long MIN_UPLOAD_PART_BYTES = 1024 * 1024L;
+
+    private final String bucketName;
+
+    private final COSClient cosClient;
+
+    private final TransferManager cosTransferManager;
+
+    public CosStorageOperator(CosStorageProperties cosStorageProperties) {
+        super(cosStorageProperties.getResourceUploadPath());
+        String secretId = cosStorageProperties.getAccessKeyId();
+        String secretKey = cosStorageProperties.getAccessKeySecret();
+        COSCredentials cosCredentials = new BasicCOSCredentials(secretId, secretKey);
+        String regionName = cosStorageProperties.getRegion();
+        ClientConfig clientConfig = new ClientConfig(new Region(regionName));
+        clientConfig.setHttpProtocol(HttpProtocol.https);
+        this.cosClient = new COSClient(cosCredentials, clientConfig);
+        this.bucketName = cosStorageProperties.getBucketName();
+        ensureBucketSuccessfullyCreated(bucketName);
+        this.cosTransferManager = getCosTransferManager();
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.cosTransferManager.shutdownNow(true);
+    }
+
+    @Override
+    public String getStorageBaseDirectory() {
+        // All directory should end with File.separator
+        if (resourceBaseAbsolutePath.startsWith("/")) {
+            log.warn("{} -> {} should not start with / in cos", Constants.RESOURCE_UPLOAD_PATH,
+                    resourceBaseAbsolutePath);
+            return resourceBaseAbsolutePath.substring(1);
```

Comment:
It's better to throw exception here since there may be abnormal data in this.

Reply:
The updated code will throw an exception and get stuck on the login page:

```java
    @Override
    public String getStorageBaseDirectory() {
        // All directory should end with File.separator
        if (resourceBaseAbsolutePath.startsWith("/")) { //  use ----> File.separator
            String errorMessage =
                    String.format("%s -> %s should not start with %s in tencent cos",
                            StorageConstants.RESOURCE_UPLOAD_PATH,
                            resourceBaseAbsolutePath, "/"); // use ----> File.separator
            throw new IllegalArgumentException(errorMessage);
        }
        return resourceBaseAbsolutePath;
    }
```

![Screenshot 2024-09-23 123421](https://github.com/user-attachments/assets/09d8f292-27f2-4cbc-8a07-6fe036ff369f)




Since there are existing test cases stating that `resource.storage.upload.base.path=/dolphinscheduler`:

![Screenshot 2024-09-23 123813](https://github.com/user-attachments/assets/91746449-27e7-4a2f-8a13-1b26c8a38253)

Throwing an exception will cause the E2E tests to fail.